plugins/aggregators/starlark/testdata/merge.star

```diff
@@ -0,0 +1,25 @@
+# Example of a merge aggregator implemented with a starlark script.
+
+load('time.star', 'time')
+
+def add(cache, metric):
+    metrics = cache.get("metrics")
+    if metrics == None:
+        metrics = {}
+        cache["metrics"] = metrics
+        cache["ordered"] = []
+    m = metrics.get(metric)
+    if m == None:
+        m = deepcopy(metric)
+        metrics[metric] = m 
+        cache["ordered"].append(m)
+    else:
+        for k, v in metric.fields.items():
+            m.fields[k] = v
```

Comment:
I'm surprised you don't need to write this back to the cache after changing it.

Reply:
It is actually what we do in [scale.star](https://github.com/influxdata/telegraf/blob/master/plugins/processors/starlark/testdata/scale.star#L12).Maybe it works because behind the scenes, it actually manipulates the pointer of the metric?